pkg/catalog/generator.go

```diff
@@ -77,11 +78,35 @@ type SourceCoder interface {
 
 // RootPath appends CurrentDir to output
 func (p *Generator) SourcePath(a SourceCoder) string {
-	rootDir := rootDirectory(path.Join(p.OutputDir, p.CurrentDir))
 	if source_path := Attribute(a, "source_path"); source_path != "" {
-		return rootDir + Attribute(a, "source_path")
+		return rootDirectory(path.Join(p.OutputDir, p.CurrentDir)) + Attribute(a, "source_path")
 	}
-	return path.Join(rootDir, a.GetSourceContext().File)
+
+	return handleSourceURL(a.GetSourceContext().File)
+}
+
+func handleSourceURL(importPath string) string {
+	//FIXME: does not work with external sysl import modules
+	//FIXME: does not work with absolute import in sysl
+	//FIXME: only handles github
+	if strings.HasPrefix(importPath, "github") {
+		urlPath, err := url.Parse("https://" + importPath)
+		if err == nil {
+			if strings.Contains(urlPath.Host, "github.") {
+				paths := strings.Split(urlPath.Path, "/")
+				// minimum length must be 3 to be a default external import
+                // user/repo/file
+				if len(paths) > 3 {
```

Comment:
Nitpick: I think this might be a little bit clearer if the check did an early return instead
e.g
```
if len(paths) < 3 {
    return path.Join(BuildGithubBlobURL(GetRemoteFromGit()), importPath)
}
```

Reply:
it's to avoid duplication of `return path.Join(BuildGithubBlobURL(GetRemoteFromGit()), importPath)`

Reply:
Okay, no worries